standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/AcidMetricService.java

```diff
@@ -85,36 +85,113 @@ public void run() {
+        TimeUnit.SECONDS)) {
+      LOG.error("A replication transaction has been open for " + metrics.getOldestOpenReplTxnAge() + " seconds. " +
+          "Before you abort a transaction that was created by replication, and which has been open a long time, " +
+          "make sure that the hive.repl.txn.timeout threshold has expired.");
+    }
     Metrics.getOrCreateGauge(NUM_OPEN_NON_REPL_TXNS).set(metrics.getOpenNonReplTxnsCount());
     Metrics.getOrCreateGauge(OLDEST_OPEN_NON_REPL_TXN_ID).set(metrics.getOldestOpenNonReplTxnId());
     Metrics.getOrCreateGauge(OLDEST_OPEN_NON_REPL_TXN_AGE).set(metrics.getOldestOpenNonReplTxnAge());
+    if (metrics.getOldestOpenNonReplTxnAge() >=
+        MetastoreConf.getTimeVar(conf, MetastoreConf.ConfVars.COMPACTOR_OLDEST_OPENTXN_THRESHOLD_WARNING,
+            TimeUnit.SECONDS)
+        && metrics.getOldestOpenNonReplTxnAge() <
+        MetastoreConf.getTimeVar(conf, MetastoreConf.ConfVars.COMPACTOR_OLDEST_OPENTXN_THRESHOLD_ERROR,
+            TimeUnit.SECONDS)) {
+      LOG.warn("Found an open transaction with an age of " + metrics.getOldestOpenNonReplTxnAge() + " seconds.");
+    } else if (metrics.getOldestOpenNonReplTxnAge() >=
+        MetastoreConf.getTimeVar(conf, MetastoreConf.ConfVars.COMPACTOR_OLDEST_OPENTXN_THRESHOLD_ERROR,
+            TimeUnit.SECONDS)) {
+      LOG.error("Found an open transaction with an age of " + metrics.getOldestOpenNonReplTxnAge() + " seconds.");
+    }
 
     Metrics.getOrCreateGauge(NUM_ABORTED_TXNS).set(metrics.getAbortedTxnsCount());
     Metrics.getOrCreateGauge(OLDEST_ABORTED_TXN_ID).set(metrics.getOldestAbortedTxnId());
     Metrics.getOrCreateGauge(OLDEST_ABORTED_TXN_AGE).set(metrics.getOldestAbortedTxnAge());
+    if (metrics.getOldestAbortedTxnAge() >=
+        MetastoreConf.getTimeVar(conf,
+            MetastoreConf.ConfVars.COMPACTOR_OLDEST_UNCLEANED_ABORTEDTXN_TIME_THRESHOLD_WARNING,
+            TimeUnit.SECONDS) &&
+        metrics.getOldestAbortedTxnAge() <
+            MetastoreConf.getTimeVar(conf,
+                MetastoreConf.ConfVars.COMPACTOR_OLDEST_UNCLEANED_ABORTEDTXN_TIME_THRESHOLD_ERROR,
+                TimeUnit.SECONDS)) {
+      LOG.warn("Found an aborted transaction with an age of " + metrics.getOldestAbortedTxnAge() + " seconds.");
+    } else if (metrics.getOldestAbortedTxnAge() >=
+        MetastoreConf.getTimeVar(conf,
+            MetastoreConf.ConfVars.COMPACTOR_OLDEST_UNCLEANED_ABORTEDTXN_TIME_THRESHOLD_ERROR,
+            TimeUnit.SECONDS)) {
+      LOG.error("Found an aborted transaction with an age of " + metrics.getOldestAbortedTxnAge() + " seconds.");
+    }
 
     Metrics.getOrCreateGauge(NUM_LOCKS).set(metrics.getLocksCount());
     Metrics.getOrCreateGauge(OLDEST_LOCK_AGE).set(metrics.getOldestLockAge());
 
     Metrics.getOrCreateGauge(TABLES_WITH_X_ABORTED_TXNS).set(metrics.getTablesWithXAbortedTxns());
+    if (metrics.getOldestAbortedTxnAge() >
```

Comment:
What do you think about saving the list of these tables in MetricsInfo, and logging that list here, so users know where to clean? It's  just an idea

Reply:
Yes, that would definitely add some value.